docker/TomcatContextUtil.pm

```diff
@@ -158,3 +227,97 @@ sub get_secret {
         die "Secret $filename empty or not defined.";
     }
 }
+
+# Return the data for a JDBC resource.
+# Fetch data using the supplied method (e.g. reading environment vars or files).
+sub get_resource {
+    my (
+        $resource,       # Name of the resource
+        $read_resource,  # Reference to method for fetching data, see get_env_var or get_resource_file
+        $password,       # Password for the resource
+        $ref             # Reference to resource id/location, to be passed to read_resource method.
+                         #   Optional. Defaults to $resource if ommitted.
+    ) = @_;
+
+    if(!defined $ref) {
+        $ref = $resource
+    }
+
+    my $url = $read_resource->($ref, 'URL', '-1');
+    my $driver;
+
+    # If environment variable MYRESOURCE_URL was set
+    if ($url ne '-1') {
+        # Detect the driver type
+        if ($url =~ /oracle:thin/) {
+            return 'oraclethin';
```

Comment:
Should this return or set $driver? If I specify a URL, will the full hash table of data ever be returned?